x-pack/plugin/core/src/main/resources/ecs-data-stream-mappings.json

```diff
@@ -0,0 +1,25 @@
+{
+  "template": {
+    "mappings": {
+      "dynamic_templates": [
+        {
+          "ecs_data_stream_to_constant": {
+            "mapping": {
+              "type": "constant_keyword"
+            },
+            "path_match": [
+              "data_stream.dataset",
+              "data_stream.namespace",
+              "data_stream.type"
```

Comment:
I'd prefer explicit mappings over dynamic templates for the data_stream fields. This guarantees that these do end up in the mapping even when the first document tries to add a bunch of fields to that the field limit is hit.

I think we can just add this directly to logs-mappings which already adds the data_stream.type field. Additionally, we shouldn't map with a nested `properties` but with a flat mapping so that logs data streams can be used with `subobjects: false`.

```
"data_stream.type": {
  "type": "constant_keyword",
  "value": "logs"
},
"data_stream.dataset": {
  "type": "constant_keyword"
},
"data_stream.namespace": {
  "type": "constant_keyword"
}
```

Reply:
Fully onboard with this.

Reply:
> > I think we can just add this directly to logs-mappings which already adds the data_stream.type field.
> 
> I like this much better! It just means that any ECS mappings users should be aware that `data_stream` fields are not included and it will make it a bit harder to detect for future ECS mapping users. @ruflin and @P1llus any objection/concerns?
> 
> > Additionally, we shouldn't map with a nested properties but with a flat mapping so that logs data streams can be used with subobjects: false.
> 
> 👍 

Sounds good to me! :)